### ISSUE TITLE
Dots after Rests

### ISSUE THREAD

**Issue by DanielS146** [MeasuringPolyphony/mp_editor#52]
In Music Input and Score Editor, dots cannot be placed after a rest.

**Comment by martha-thomae** [comment]
I am under the impression that Verovio is still not rendering dots after rests, but I don't know if this could be related to this issue. Could this be the case, @JRegimbal ?

**Comment by JRegimbal** [comment]
Sorry for the delay. It's easy to allow dots to be added to rests and by default it is converted to MEI as `dots=1` added to the rest element. @martha-thomae is correct that Verovio does not render this, so I'm unsure if allowing essentially invisible dots would be more trouble than its worth.

**Comment by JRegimbal** [comment]
(Also looking at the VHV using the separate dot element doesn't make a difference in anything being rendered.)

**Comment by martha-thomae** [comment]
> (Also looking at the VHV using the separate dot element doesn't make a difference in anything being rendered.)

It probably doesn't. But in Mensural MEI the "right way" to encode a dot is with a separate element, and not as an attribute (which is what you would do in CMN). What happens if you remove the attribute and leave the element, does it stops rendering dots altogether?

**Comment by martha-thomae** [comment]
> Sorry for the delay. It's easy to allow dots to be added to rests and by default it is converted to MEI as `dots=1` added to the rest element. @martha-thomae is correct that Verovio does not render this, so I'm unsure if allowing essentially invisible dots would be more trouble than its worth.

I haven't tried this (using `@dots` in `<rest>`). 

I know it didn't render 
```xml
<rest/>
<dot/>
```
I don't know if it does now.

**Comment by JRegimbal** [comment]
Checking now, it sort of does? It doesn't show up right reading from humdrum (i.e., in the input editor) but it does appear with the position very off when reading from MEI where dot after rest can be specified. However it seems to just end up getting placed closer to a preceding note element rather than after the rest itself.

**Comment by martha-thomae** [comment]
Yes, that last part sounds familiar. I think I had this issue before (Verovio rendering the dot of the rest on the preceding / following note).

**Comment by karend27** [comment]
So I'm looking at an ars antiqua piece we encoded for the Measuring Polyphony project that has dots after rests - screenshot here and mensural mei file attached. The dot is placed here where it should be: right after the rest.
<img width="760" alt="Screen Shot 2020-09-04 at 10 22 36 AM" src="https://user-images.githubusercontent.com/11195203/92249903-c83b7180-ee98-11ea-81fd-805d17a6ebcf.png">
This is the code snippet for the triplum voice at that point:
`<note xml:id="m-894" dur="brevis" oct="3" oct.ges="2" pname="g">
									<verse xml:id="m-905" n="1">
										<syl xml:id="m-906" wordpos="t">te</syl>
									</verse>
								</note>
								<note xml:id="m-895" dur="semibrevis" oct="3" oct.ges="2" pname="a">
									<verse xml:id="m-907" n="1">
										<syl xml:id="m-908" wordpos="i">a</syl>
									</verse>
								</note>
								<rest xml:id="m-896" dur="semibrevis" dur.quality="major" num="1" numbase="2" />
								<dot/>
								<note xml:id="m-897" dur="semibrevis" oct="4" oct.ges="3" pname="c">
									<verse xml:id="m-909" n="1">
										<syl xml:id="m-910" wordpos="i">la</syl>
									</verse>
								</note>
								<rest xml:id="m-898" dur="semibrevis" dur.quality="major" num="1" numbase="2" />
								<dot/>
								<barLine form="dashed" xml:id="m-f75de090-253b-4586-95d0-e0a1272aedb1" />
								<note xml:id="m-935" dur="semibrevis" oct="3" oct.ges="2" pname="a">
									<verse xml:id="m-942" n="1">
										<syl xml:id="m-943" con="d" wordpos="i">bou</syl>
									</verse>
								</note>
								<note xml:id="m-936" dur="semibrevis" oct="3" oct.ges="2" pname="g" dur.quality="major" num="1" numbase="2">
									<verse xml:id="m-944" n="1">
										<syl xml:id="m-945" wordpos="t">cete</syl>
									</verse>
								</note>
								<dot xml:id="m-1e22f33b-8e8a-47e7-8ab9-6c69f786bf27" />
								
[332_MENSURAL.txt](https://github.com/MeasuringPolyphony/mp_editor/files/5175407/332_MENSURAL.txt)

`

**Comment by martha-thomae** [comment]
Hmmm, you are right, they are rendering correctly. Let me check, I am pretty sure I was working with larger note values. Will check if there is any difference.

**Comment by martha-thomae** [comment]
Yes, it seems to be rendered in the right place for smaller values of rests than larger. Here is an MEI file and the screenshot of its rendition in Verovio. The dots are encoded always after the rest in the MEI file, but they are rendered before them in Verovio for larger note values.

[notes_rests_dots.mei.zip](https://github.com/MeasuringPolyphony/mp_editor/files/5175773/notes_rests_dots.mei.zip)

![Verovio-rendering](https://user-images.githubusercontent.com/13948831/92258338-ac3dcd00-eea4-11ea-97af-cb762c375be3.png)

**Comment by karend27** [comment]
Will I raise this as an issue in the Verovio github?

**Comment by martha-thomae** [comment]
Yes, I think that might be best.
Please feel free to use any of the files and images I attached :)

**Comment by karend27** [comment]
Ok - I opened a new issue here: https://github.com/rism-ch/verovio/issues/1682
